crates/sui-core/src/signature_mpc/mod.rs

```diff
@@ -340,33 +343,50 @@ impl SignatureMPCAggregator {
             SignatureMPCMessageProtocols::Sign(m) => {
                 let mut state = sign_session_states.entry(session_id).or_insert_with(|| {
                     SignState::new(
+                        tiresias_key_share_decryption_key_share,
                         tiresias_public_parameters.clone(),
                         epoch,
                         party_id,
-                        parties,
+                        parties.clone(),
                         session_id,
                     )
                 });
-
                 let _ = state.insert_first_round(sender_party_id, m.clone());
-
                 if let Some(r) = sign_session_rounds.get_mut(&session_id) {
                     if state.ready_for_complete_first_round(&r) {
                         drop(r);
-                        let state = state.clone();
```

Comment:
why did you move it inside?

Reply:
We changed it because, in this case, we needed a mutable reference to the state. If the signature verification fails, we need to store the indices of the failed messages and the IDs of the involved parties in the state. When we tried to pass the mutable reference from outside, we encountered lifetime errors.